controllers/capabilities/developeraccount_threescale_reconciler.go

```diff
@@ -39,21 +39,48 @@ func NewDeveloperAccountThreescaleReconciler(b *reconcilers.BaseReconciler, reso
 func (s *DeveloperAccountThreescaleReconciler) Reconcile() (*threescaleapi.DeveloperAccount, error) {
 	s.logger.V(1).Info("START")
 
-	// Reconciliation is based on ID stored in Status field
-	// All fields of the spec are not unique
+	// Reconciliation is based on the ID stored in the CR's annotation or .status block
+	// This is required because none of the fields in the DeveloperAccount CR's .spec are unique
 	// For instance, there may exist several DevAccounts with the same Organization Name.
-	// The only way to know that the account is already created in 3scale is checking the ID in status.
-	// Nice to Have would be having that ID in status inmutable using admission webhooks
-	devAccount, err := s.findDevAccountByID()
+	// We can tell if the account is already created in 3scale by checking the ID in CR's annotation or .status block
+	accountId, err := s.retrieveAccountID()
+	if err != nil {
+		return nil, err
+	}
+
+	devAccount, err := s.findDevAccountByID(accountId)
 	if err != nil {
 		return nil, err
 	}
 
 	if devAccount == nil {
 		s.logger.V(1).Info("DeveloperAccount does not exist", "OrgName", s.resource.Spec.OrgName)
-		// ID not in status field
-		// developer account has to be created in 3scale
-		return s.createDevAccount()
+		// The DeveloperAccount doesn't exist yet and must be created in 3scale
+		createdDevAccount, createErr, devUserCR := s.createDevAccount()
+		if createErr != nil {
+			// Occasionally system will return a 409 error even though the DeveloperAccount was created successfully.
+			// When this happens, the returned createdDevAccount will have nil values for all its fields.
+			// If this happens the operator should be able to recover and not mark the DeveloperAccount CR as failed.
+			// Otherwise, the operator will try to create the same DA again which will produce a real error.
+			portaErr, ok := createErr.(threescaleapi.ApiErr)
+			if ok && portaErr.Code() == 409 {
```

Comment:
did we managed to reproduce and test it with the PR?

Reply:
Yes I was able to reproduce it consistently and verify that this `if` statement handles it properly